### PR TITLE
chore: change Array.forEach to for...of in JS function templates

### PR DESCRIPTION
**Pull request by josefaidt**
#### Description of changes

changes use of `Array.forEach` to `for..of` in DynamoDB Node.js function templates

#### Issue #, if available

ref #10969 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-nodejs-function-template-provider/resources/lambda/trigger/trigger-dynamodb.js.ejs [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 exports.handler = event => {
   console.log(`EVENT: ${JSON.stringify(event)}`);
-  event.Records.forEach(record => {
+  for (const record of event.Records) {
```
Comment (jhockett) [inline packages/amplify-nodejs-function-template-provider/resources/lambda/trigger/trigger-dynamodb.js.ejs:8]: There's a closing `)` that needs to be removed too.

**Review by jhockett — CHANGES_REQUESTED** [review]

File: packages/amplify-nodejs-function-template-provider/resources/lambda/trigger/trigger-kinesis.js.ejs [hunk 2]
```diff
@@ -7,7 +7,7 @@ exports.handler = event => {
   console.log(`EVENT: ${JSON.stringify(event)}`);
   let res = '';
   if ('Records' in event) {
-    event.Records.forEach(record => {
+    for (const record of event.Records) {
```
Comment (jhockett) [inline packages/amplify-nodejs-function-template-provider/resources/lambda/trigger/trigger-kinesis.js.ejs:10]: Same here, line 14.

File: packages/amplify-nodejs-function-template-provider/resources/lambda/trigger/trigger-dynamodb.js.ejs (hunk 1 above)
Comment (josefaidt) [reply packages/amplify-nodejs-function-template-provider/resources/lambda/trigger/trigger-dynamodb.js.ejs:8]: ah good call, removed

**Review by jhockett — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Comment by danielleadams** [comment]
@josefaidt it looks like `yarn test` is failing

**Comment by josefaidt** [comment]
I'm seeing tests failing locally as well, but in a seemingly unrelated package. Pushed a merge to see if that helps

**Pull request merged**